### PR TITLE
Feature: Implement centralised logger

### PR DESCRIPTION
**Pull request by SandPod**
Issue: https://github.com/serverpod/serverpod/issues/1042

Introduces a centralised logger as a first step of improving the overall CLI output.

The purpose of the centralised logger is to make it easier to implement and switch out other logging solutions based on platform demands or available libraries.

Adding the interface is done in isolation so that the migration work doesn't block any work.

There will be a follow up PR that moves all existing print statement to the logger.

#### Error message
Before change in blue and after change in red:
Screenshot 2023-07-10 at 11 51 50

#### Warning message
Before change in blue and after change in red:

Screenshot 2023-07-10 at 11 54 58

## Pre-launch Checklist

- [x] I read the [Contribute](https://docs.serverpod.dev/contribute) page and followed the process outlined there for submitting PRs.
- [x] This update contains only one single feature or bug fix and nothing else. (If you are submitting multiple fixes, please make multiple PRs.)
- [x] I read and followed the [Dart Style Guide](https://dart.dev/guides/language/effective-dart/style) and formatted the code with [dart format](https://dart.dev/tools/dart-format).
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`), and made sure that the documentation follows the same style as other Serverpod documentation. I checked spelling and grammar.
- [ ] I added new tests to check the change I am making.
- [ ] All existing and new tests are passing.
- [ ] Any breaking changes are documented below.

If you need help, consider asking for advice on the [discussion board](https://github.com/serverpod/serverpod/discussions).

### CODE REVIEW

File: tools/serverpod_cli/lib/src/logger/logger.dart [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/// Serverpods internal logger interface.
+/// All logging output should go through this interface.
+/// The purpose is to simplify implementing and switching out concrete logger
+/// implementations.
+abstract class Logger {
+  /// Display a [message] wrapped in a box.
+  /// Commands should use this when they want to highlight a message that otherwise is easily missed.
+  void printBox(String message);
```
Comment (SandPod) [inline tools/serverpod_cli/lib/src/logger/logger.dart:8]: This feels misplaced.

This class more communicates the log level of the message while this method communicates a style. And error might want to have a box as well.

Should we remove this and add it as a style option for Info for now.

File: tools/serverpod_cli/bin/serverpod_cli.dart [hunk 2]
```diff
@@ -38,6 +40,8 @@ final runModes = <String>['development', 'staging', 'production'];
 final Analytics _analytics = Analytics();
 
 void main(List<String> args) async {
+  logger = StdOutLogger();
```
Comment (SandPod) [inline tools/serverpod_cli/bin/serverpod_cli.dart:43]: This solution has a critical problem.

When running tests there will be no logger initialised.

Therefore we can't rely on a s final singleton initialised in the bin file.

File: tools/serverpod_cli/lib/src/logger/logger.dart (hunk 1 above)
Comment (vlidholt) [reply tools/serverpod_cli/lib/src/logger/logger.dart:8]: A style option sounds like a better choice. I think we should have the same style options for all log levels.

File: tools/serverpod_cli/bin/serverpod_cli.dart (hunk 2 above)
Comment (vlidholt) [reply tools/serverpod_cli/bin/serverpod_cli.dart:43]: Isn't this what the `setUp` method in the test is for?

**Review by vlidholt — CHANGES_REQUESTED** [review]
I would want to see this a bit more completed before we merge it. E.g. options for drawing the boxes and for word wrap, etc.

**Pull request closed without merging**